torch/_meta_registrations.py

```diff
@@ -1472,18 +1472,25 @@ def meta_scatter_add(self, dim, index, src):
 @register_meta(aten.upsample_nearest2d.vec)
 def upsample_nearest2d_vec(input, output_size, scale_factors):
     mem_format = utils.suggest_memory_format(input)
+    spatial_dimensions = input.dim() - 2
+
+    input_shape = input.shape
     if output_size is not None:
         assert scale_factors is None
-        return input.new_empty(input.size()).to(memory_format=mem_format)
-
-    assert output_size is None
-    out_size = list(input.size())
-
-    for i in range(len(out_size) - 2):
-        sym_float = (out_size[i + 2] / 1) * scale_factors[i]
-        assert sym_float >= 0
-        out_size[i + 2] = math.floor(sym_float)
-    return input.new_empty(out_size).to(memory_format=mem_format)
+        out_size = output_size
+    elif scale_factors is not None:
+        assert output_size is None
+        out_size = []
+        for i in range(spatial_dimensions):
+            sym_float = (input_shape[i + 2] / 1) * scale_factors[i]
+            assert sym_float >= 0
+            out_size.append(math.floor(sym_float))
+
+    output_height = out_size[0]
+    output_width = out_size[1]
+    nbatch = input_shape[0]
+    channels = input_shape[1]
+    return input.new_empty((nbatch, channels, output_height, output_width)).to(memory_format=mem_format)
```

Comment:
cc @Krovatkin 

tbh I just started at the C++ meta kernel defined in `UpSampleNearest2d.cpp` to get this change, and it passed that failing `test/test_meta.py` test